src/metaArrow.cxx

```diff
@@ -50,7 +50,7 @@ MetaArrow::MetaArrow(unsigned int dim)
 {
   META_DEBUG_PRINT( "MetaArrow()" );
   MetaArrow::Clear();
-  m_NDims = dim;
+  MetaObject::InitializeEssential(dim);
```

Comment:
Why is the arrow the only one to receive special updates (in addition to image)?

Reply:
They are the only ones which were inconsistent.   Tubes, blobs, boxes, contours, etc already existed in their own coordinate systems.  Arrow only had a length and a direction prior to this.  The object-to-parent transform had to be used to move the arrow to a position.  It really made handling annotations inconsistent.   This change is conceptually very helpful - particularly for maskImages and images.  I swear this all worked correctly years ago when Julien and I wrote spatial object viewers, but I haven't tried to go back and look at where things went wonky between metaIO and spatialObjects.

Reply:
And please don't merge yet - note I changed name to WIP.

I found a memory issue when handling user-defined fields.   It was very brittle - it worked if you called the exact right functions in the exact right order.  I've updated to make it much more logical and robust.   Will include another commit in this pull request.  Also, will rebase and squash all before merging.

Thanks!

Reply:
I believe you can convert the PR to `Draft` status.